src/AsyncTCP.cpp

```diff
@@ -856,6 +973,22 @@ int8_t AsyncClient::_sent(tcp_pcb* pcb, uint16_t len) {
 int8_t AsyncClient::_recv(tcp_pcb* pcb, pbuf* pb, int8_t err) {
     while(pb != NULL) {
         _rx_last_packet = millis();
+
+#if ASYNC_TCP_SSL_ENABLED
+        if(_pcb_secure){
+            // log_i("_recv: %d\n", pb->tot_len);
+            int read_bytes = tcp_ssl_read(pcb, pb);
+            if(read_bytes < 0){
+                if (read_bytes != MBEDTLS_ERR_SSL_PEER_CLOSE_NOTIFY) {
+                    log_e("_recv err: %d\n", read_bytes);
+                    _close();
+                }
+                return ERR_BUF; // for lack of a better error value
+            }
+            return ERR_OK;
```

Comment:
you are in a while loop. should continue here maybe?

Reply:
fixed